IRCT-RI/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/ri/i2b2/I2B2XMLResourceImplementation.java

```diff
@@ -899,134 +902,145 @@ private void 	convertPatientDataResponseTypeToResultSet(PatientDataResponseType
-		Map<String, Map<String,String>> whateverStorage = new HashMap<>();
- 		for (ObservationSet observationSet : observationSetList){
-
- 			for (ObservationType observationType : observationSet.getObservation()){
- 				String patientId = observationType.getPatientId().getValue();
- 				String columnName = (conceptCD_aliasName_Map.containsKey(observationType.getConceptCd().getValue()))?
-						conceptCD_aliasName_Map.get(observationType.getConceptCd().getValue())
-						:observationType.getConceptCd().getValue();
-
-
- 				// handle where to retrieve the value
-				// several cases:
-				// 1. value type is T (means text), Tval has data
-				// 2. value type is N (means number), Nval has data
-				// 3. value type is null (maybe value is not a observation fact), Tval, Nval both are null
-				String value = "";
-				String valueType = observationType.getValuetypeCd();
-				if (valueType == null || valueType.equals("@")){
-					value = observationType.getConceptCd().getValue(); // like Gender, Age something
-				} else if (valueType.equals("T")){
-					value = observationType.getTvalChar();
-				} else if (valueType.equals("N")){
-					value = observationType.getNvalNum().getValue().toPlainString();
+					if (whateverStorage.containsKey(patientId)) {
+						whateverStorage.get(patientId)
+								.put(columnName, value);
+					} else {
+						Map<String, String> temp = new HashMap<>();
+						temp.put(columnName, value);
+						whateverStorage.put(patientId, temp);
+					}
 				}
-				if (value == null)
-					value = "";
+			}
 
- 				if (whateverStorage.containsKey(patientId)) {
- 					whateverStorage.get(patientId)
-							.put(columnName, value);
-				} else {
- 					Map<String, String> temp = new HashMap<>();
- 					temp.put(columnName, value);
- 					whateverStorage.put(patientId, temp);
+			// ok... now start to append rows... the whole thing is terrible... anyways... too many words... get things done
+			for (Map.Entry<String, Map<String, String>> entry : whateverStorage.entrySet()){
+				mrs.appendRow();
+				mrs.updateString("Patient Id", entry.getKey());
+				for (Map.Entry<String, String> innerEntry : entry.getValue().entrySet()){
+					mrs.updateString(innerEntry.getKey(), innerEntry.getValue());
 				}
 			}
-		}
-
-		// ok... now start to append rows... the whole thing is terrible... anyways... too many words... get things done
-		for (Map.Entry<String, Map<String, String>> entry : whateverStorage.entrySet()){
-			mrs.appendRow();
-			mrs.updateString("Patient Id", entry.getKey());
-			for (Map.Entry<String, String> innerEntry : entry.getValue().entrySet()){
-				mrs.updateString(innerEntry.getKey(), innerEntry.getValue());
+			} else {
+			for (PidType ps : patientDataResponse.getPatientData().getPidSet().getPid()){
+				if (this.sourceWhiteList.contains(ps.getPatientId().getSource())){
+					mrs.appendRow();
+					mrs.updateString("Patient Id", ps.getPatientId().getValue());
```

Comment:
see screenshots: <img width="856" alt="image" src="https://user-images.githubusercontent.com/33844814/40081170-36563b7c-585b-11e8-8a3d-3deb3c0fa748.png">
which one should we look into?

Reply:
direct from i2b2 org about PID:
![image](https://user-images.githubusercontent.com/33844814/40081207-56242892-585b-11e8-8eef-566a4dec1723.png)

Reply:
we really need both to come back, so you would pseu-do something like this:

select: Hospital-1 and Hospital-5
where: patients have lab results of some kind

and get something like this in return:

patient_id, Hospital-1, Hospital-5
100000001, 2000001961, 3000001821
...

Reply:
So instead of one column of just the patient id, we need multiple columns, where the heading of the columns is the patient_map_id source??  Meaning multiple columns with the same heading?  This needs a lot more explanation, and to be in the ticket.

Reply:
We are clear now 👍

Reply:
Yep!  All updated.